lib/x86simdsort.cpp

```diff
@@ -10,7 +10,8 @@ static int check_cpu_feature_support(std::string_view cpufeature)
     const char *disable_avx512 = std::getenv("XSS_DISABLE_AVX512");
 
     if ((cpufeature == "avx512_spr") && (!disable_avx512))
-#if defined(__FLT16_MAX__) && !defined(__INTEL_LLVM_COMPILER)
+#if defined(__FLT16_MAX__) && !defined(__INTEL_LLVM_COMPILER) && \
+      !defined(__OpenBSD__)
```

Comment:
Could you change this to a conditional based on clang version >=18? clang++ v18 supports `__builtin_cpu_supports("avx512fp16")`. See https://godbolt.org/z/aWrGfGsqY